java-desktop/src/main/java/com/couchbase/todo/TodoApp.java

```diff
@@ -35,34 +45,35 @@ public void stop() throws Exception {
         super.stop();
     }
 
-    public static void gotoLoginScreen(Stage stage) {
-        try {
-            FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource("/scene/Login.fxml"));
+    public static void setScene(Stage stage, String fxmlFile) {
+        FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource(fxmlFile));
+
+        if (fxmlFile.equals(MAIN_FXML)) {
+            loader.setController(new MainController(stage));
+        }
+        else if (fxmlFile.equals(LOGIN_FXML)) {
             loader.setController(new LoginController(stage));
-            Parent root = loader.load();
-            Scene scene = new Scene(root);
-            stage.setScene(scene);
         }
-        catch (IOException e) {
-            throw new IllegalStateException(e);
+        else if (fxmlFile.equals(CONFIG_FXML)) {
+            loader.setController(new ConfigController(stage));
         }
+
+        Parent root;
+        try { root = loader.load(); }
+        catch (IOException e) { throw new RuntimeException("IOException when set scene to : " + fxmlFile, e); }
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
     }
 
-    public static void gotoMainScreen(Stage stage) {
-        try {
-            FXMLLoader loader = new FXMLLoader(TodoApp.class.getResource("/scene/main.fxml"));
-            loader.setController(new MainController(stage));
-            Parent root = loader.load();
-            Scene scene = new Scene(root);
-            stage.setScene(scene);
-        }
-        catch (IOException e) {
-            throw new IllegalStateException(e);
-        }
+    public static synchronized Config getConfig() {
```

Comment:
So... synchronized *and* volatile is a belt and suspenders.  You don't nee both.  Both are somewhat expensive, so it is slightly better if you can avoid them.
I think what you are attempting is to guarantee that getConfig never returns null.  That's a great idea.
I think that what you want, here, though, is to use an AtomicReference (I only say this after looking at your code: I guessed that a volatile would be enough)  That will get rid of both the sync and the volatile.
You want code like this:
`
config == this.config.get();
if (config != null) { return config; } 
config == new Config.Builder().build();
this.config.compareAndSet(null, config);
return this.config.get();
`